plugins/imagebase/plugin.js

```diff
@@ -104,6 +272,46 @@
 			var widget = editor.widgets.add( name, createWidgetDefinition( editor, definition ) );
 
 			editor.addFeature( widget );
+		},
+
+		/**
+		 * Adds new feature to the newly created image widget by invoking initial setup once for the editor
+		 * and extending widget's definition to include all fields needed by this feature.
+		 *
+		 *		var widgetDefinition = {};
+		 *		widgetDefinition = CKEDITOR.plugins.imagebase.addFeature( editor, 'link', widgetDefinition );
+		 *		CKEDITOR.plugins.imagebase.addImageWidget( editor, 'myWidget', widgetDefinition );
+		 *
+		 * @param {CKEDITOR.editor} editor Editor that will get the widget registered.
+		 * @param {String} name Feature name.
+		 * @param {CKEDITOR.plugins.imagebase.imageWidgetDefinition} definition Widget's definition.
+		 * @returns {CKEDITOR.plugins.imagebase.imageWidgetDefinition} Widget's definition extended
+		 * with fields needed by feature.
+		 */
+		addFeature: function( editor, name, definition ) {
+			var featureDefinition = CKEDITOR.tools.clone( this.featuresDefinitions[ name ] );
+
+			function mergeMethods( oldOne, newOne ) {
+				if ( !oldOne && !newOne ) {
+					return;
+				}
+
+				return function() {
+					oldOne && oldOne.apply( this, arguments );
+					newOne && newOne.apply( this, arguments );
+				};
+			}
+
+			featureDefinition.init = mergeMethods( definition.init, featureDefinition.init );
+			featureDefinition.data = mergeMethods( definition.data, featureDefinition.data );
+
+			if ( featureDefinition.setUp ) {
+				featureDefinition.setUp( editor );
```

Comment:
If this function is meant to be a candidate for widget feature `setUp` function, you should also pass `definition` to it, so that widget feature has a possibility to have a look at widget that attempts to reuse it.